File: snippets/fsharp/System/TimeSpan/Parse/parse1.fs
module parse1

// <Snippet1>
open System
open System.Globalization
open System.Threading

let values = 
    [| "6"; "6:12"; "6:12:14"; "6:12:14:45" 
       "6.12:14:45"; "6:12:14:45.3448" 
       "6:12:14:45,3448"; "6:34:14:45" |]
let cultureNames = [| "hr-HR"; "en-US" |]

// Change the current culture.
for cultureName in cultureNames do
    Thread.CurrentThread.CurrentCulture <- CultureInfo cultureName
    printfn $"Current Culture: {Thread.CurrentThread.CurrentCulture.Name}" 
    for value in values do
        try 
            let ts = TimeSpan.Parse value
            printfn $"{value} --> {ts:c}"
        with 
        | :? FormatException ->
            printfn $"{value}: Bad Format"
        | :? OverflowException ->
            printfn $"{value}: Overflow"
    printfn ""                                
// The example displays the following output:
//    Current Culture: hr-HR
//    6 --> 6.00:00:00
//    6:12 --> 06:12:00
//    6:12:14 --> 06:12:14
//    6:12:14:45 --> 6.12:14:45
//    6.12:14:45 --> 6.12:14:45
//    6:12:14:45.3448: Bad Format
//    6:12:14:45,3448 --> 6.12:14:45.3448000
//    6:34:14:45: Overflow
//    
//    Current Culture: en-US
//    6 --> 6.00:00:00
//    6:12 --> 06:12:00
//    6:12:14 --> 06:12:14
//    6:12:14:45 --> 6.12:14:45
//    6.12:14:45 --> 6.12:14:45
//    6:12:14:45.3448 --> 6.12:14:45.3448000
//    6:12:14:45,3448: Bad Format
//    6:34:14:45: Overflow
// </Snippet1>

File: snippets/csharp/System/UInt32/Parse/parse1.cs
using System;

public class ParseExample1
{
    public static void Run()
    {
        // <Snippet1>
        string[] values = { "+13230", "-0", "1,390,146", "$190,235,421,127",
                          "0xFA1B", "163042", "-10", "2147483648",
                          "14065839182", "16e07", "134985.0", "-12034" };
        foreach (string value in values)
        {
            try
            {
                uint number = uint.Parse(value);
                Console.WriteLine($"{value} --> {number}");
            }
            catch (FormatException)
            {
                Console.WriteLine($"{value}: Bad Format");
            }
            catch (OverflowException)
            {
                Console.WriteLine($"{value}: Overflow");
            }
        }
        // The example displays the following output:
        //       +13230 --> 13230
        //       -0 --> 0
        //       1,390,146: Bad Format
        //       $190,235,421,127: Bad Format
        //       0xFA1B: Bad Format
        //       163042 --> 163042
        //       -10: Overflow
        //       2147483648 --> 2147483648
        //       14065839182: Overflow
        //       16e07: Bad Format
        //       134985.0: Bad Format
        //       -12034: Overflow
        // </Snippet1>
    }
}
